File: src/bot/keyboards/main.keyboard.ts
/**
 * Main Keyboard
 * Primary navigation keyboard for registered users
 */

import { Markup } from 'telegraf';
import { BUTTON_LABELS } from '../../utils/constants';

/**
 * Get main menu keyboard
 * @param isAdmin - Whether user is admin
 */
export const getMainKeyboard = (isAdmin: boolean = false) => {
  const buttons = [
    [
      Markup.button.callback(BUTTON_LABELS.PROFILE, 'profile'),
      Markup.button.callback(BUTTON_LABELS.DEPOSITS, 'deposits'),
    ],
    [
      Markup.button.callback(BUTTON_LABELS.WITHDRAWALS, 'withdrawals'),
      Markup.button.callback(BUTTON_LABELS.REFERRALS, 'referrals'),
    ],
    [
      Markup.button.callback(BUTTON_LABELS.TRANSACTIONS, 'transaction_history'),
    ],
    [
      Markup.button.callback(BUTTON_LABELS.SUPPORT, 'support'),
    ],
    [
      Markup.button.callback(BUTTON_LABELS.HELP, 'help'),
    ],
  ];

  // Add admin panel button for admins
  if (isAdmin) {
    buttons.push([
      Markup.button.callback(BUTTON_LABELS.ADMIN_PANEL, 'admin_panel'),
    ]);
  }

  return Markup.inlineKeyboard(buttons);
};

/**
 * Get welcome keyboard for new users
 */
export const getWelcomeKeyboard = () => {
  return Markup.inlineKeyboard([
    [Markup.button.callback(BUTTON_LABELS.START_REGISTRATION, 'start_registration')],
  ]);
};

export default {
  getMainKeyboard,
  getWelcomeKeyboard,
};

File: bot/keyboards/main_keyboard.py
"""
Main Keyboard
Main menu keyboard for the bot
"""

from aiogram.types import InlineKeyboardMarkup, InlineKeyboardButton


def get_main_keyboard(is_admin: bool = False) -> InlineKeyboardMarkup:
    """
    Get main menu keyboard

    Args:
        is_admin: Whether the user is an admin

    Returns:
        Main menu keyboard
    """
    buttons = [
        [
            InlineKeyboardButton(
                text="💰 Депозит", callback_data="deposit"
            ),
            InlineKeyboardButton(
                text="💸 Вывод", callback_data="withdrawal"
            ),
        ],
        [
            InlineKeyboardButton(
                text="🤝 Рефералы", callback_data="referrals"
            ),
            InlineKeyboardButton(
                text="👤 Профиль", callback_data="profile"
            ),
        ],
        [
            InlineKeyboardButton(
                text="📊 История", callback_data="transaction_history"
            ),
            InlineKeyboardButton(
                text="🆘 Поддержка", callback_data="support"
            ),
        ],
    ]

    # Add admin panel button if user is admin
    if is_admin:
        buttons.append(
            [
                InlineKeyboardButton(
                    text="👑 Админ-панель", callback_data="admin_panel"
                )
            ]
        )

    return InlineKeyboardMarkup(inline_keyboard=buttons)
